galasa-extensions-parent/dev.galasa.auth.couchdb/src/main/java/dev/galasa/auth/couchdb/internal/CouchdbAuthStore.java

```diff
@@ -0,0 +1,120 @@
+import dev.galasa.extensions.common.api.LogFactory;
+import dev.galasa.extensions.common.couchdb.CouchdbException;
+import dev.galasa.extensions.common.couchdb.CouchdbStore;
+import dev.galasa.extensions.common.couchdb.CouchdbValidator;
+import dev.galasa.extensions.common.couchdb.pojos.ViewRow;
+import dev.galasa.extensions.common.api.HttpRequestFactory;
+import dev.galasa.framework.spi.auth.IAuthToken;
+import dev.galasa.framework.spi.auth.IAuthStore;
+import dev.galasa.framework.spi.auth.User;
+import dev.galasa.framework.spi.utils.ITimeService;
+import dev.galasa.framework.spi.auth.AuthStoreException;
+
+/**
+ * When CouchDB is being used to store user-related information, including authentication
+ * tokens, this class is called upon to implement the auth store. This class registers the
+ * auth store as the only auth store in the framework, and is only used when Galasa is
+ * running in an ecosystem.
+ *
+ * This implementation of the auth store interface gets all of its data from a CouchDB
+ * server.
+ */
+public class CouchdbAuthStore extends CouchdbStore implements IAuthStore {
+
+    public static final String TOKENS_DATABASE_NAME = "galasa_tokens";
+    public static final String COUCHDB_AUTH_ENV_VAR = "GALASA_AUTHSTORE_TOKEN";
+    public static final String COUCHDB_AUTH_TYPE    = "Basic";
+
+    private Log logger;
+    private ITimeService timeService;
+
+    public CouchdbAuthStore(
+        URI authStoreUri,
+        HttpClientFactory httpClientFactory,
+        HttpRequestFactory requestFactory,
+        LogFactory logFactory,
+        CouchdbValidator validator,
+        ITimeService timeService
+    ) throws CouchdbException {
+        super(authStoreUri, requestFactory, httpClientFactory);
+        this.logger = logFactory.getLog(getClass());
+        this.timeService = timeService;
+
+        validator.checkCouchdbDatabaseIsValid(this.storeUri, this.httpClient, this.httpRequestFactory);
+    }
+
+    @Override
+    public List<IAuthToken> getTokens() throws AuthStoreException {
+        logger.info("Retrieving tokens from CouchDB");
+        List<ViewRow> tokenDocuments = new ArrayList<>();
+        List<IAuthToken> tokens = new ArrayList<>();
+
+        try {
+            // Get all of the documents in the tokens database
+            tokenDocuments = getAllDocsFromDatabase(TOKENS_DATABASE_NAME);
+
+            // Build up a list of all the tokens using the document IDs
+            for (ViewRow row : tokenDocuments) {
+                tokens.add(getAuthTokenFromDocument(row.key));
+            }
+
+            logger.info("Tokens retrieved from CouchDB OK");
+        } catch (CouchdbException e) {
+            String errorMessage = ERROR_FAILED_TO_RETRIEVE_TOKENS.getMessage(e.getMessage());
+            throw new AuthStoreException(errorMessage, e);
+        }
+        return tokens;
+    }
+
+    @Override
+    public void shutdown() throws AuthStoreException {
+        try {
+            httpClient.close();
```

Comment:
Wondering whether a test pod will instantiate this extension or not. Again, back to only API layer needing the auth service store.